app/services/mentor_at_school_periods/finish.rb

```diff
@@ -12,6 +13,14 @@ def finish_existing_at_school_periods!
         period.update!(finished_on:)
         period.mentorship_periods.ongoing_on(finished_on).update!(finished_on:)
         period.training_periods.ongoing_on(finished_on).update!(finished_on:)
+
+        Events::Record.record_teacher_finishes_mentor_at_school_period_event!(
```

Comment:
For consistency with the ECT variant, maybe `#record_teacher_left_school_as_ect!` would be a better fit?